src/ray/raylet/node_manager.cc

```diff
@@ -1575,7 +1583,19 @@ void NodeManager::DispatchScheduledTasksToWorkers() {
 
 void NodeManager::NewSchedulerSchedulePendingTasks() {
   RAY_CHECK(new_scheduler_enabled_);
-  while (!tasks_to_schedule_.empty()) {
+  size_t queue_size = tasks_to_schedule_.size();
+
+  // Check every task in task_to_schedule queue to see
+  // whether it can be scheduled. This avoids head-of-line
+  // blocking where a task which cannot be scheduled because
+  // there are not enough available resources blocks other
+  // tasks from being scheduled.
+  while (queue_size > 0) {
+    if (queue_size == 0) {
```

Comment:
This condition is unreachable - the while loop would break

Reply:
Could you use a for loop instead? `for (size_t queue_size = tasks_to_schedule_.size(); queue_size > 0; queue_size--)`